packages/components/src/internal/components/picklist/PicklistDeleteConfirm.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { FC, memo, ReactNode, useCallback, useEffect, useMemo, useState } from 'react';
+import { getPicklistDeleteData, PicklistDeletionData } from './actions';
+import { PicklistModel } from './models';
+import { QueryModel } from '../../../public/QueryModel/QueryModel';
+import { Actions } from '../../../public/QueryModel/withQueryModels';
+import { User } from '../base/models/User';
+import { Alert } from '../base/Alert';
+import { ConfirmModal } from '../base/ConfirmModal';
+import { LoadingSpinner } from '../base/LoadingSpinner';
+
+
+interface Props {
+    model: QueryModel,
+    actions: Actions,
+    user: User,
+    useSelection: boolean,
+    onConfirm: (listsToDelete: any[]) => void,
+    onCancel: () => void
+}
+
+export const PicklistDeleteConfirm: FC<Props> = memo(props => {
+
+    const { model, actions, onConfirm, onCancel, useSelection, user } = props;
+    const [ errorMessage, setErrorMessage ] = useState(undefined);
+    const [ nounAndNumber, setNounAndNumber ] = useState('Picklist');
+    const [ deletionData, setDeletionData ]  = useState<PicklistDeletionData>(undefined);
+
+    const numSelected = useSelection ? model.selections.size : 1;
+    const noun = (numSelected === 1) ? 'Picklist' : 'Picklists';
+    const lcNoun = noun.toLowerCase();
+
+    useEffect(() =>  {
+        if (useSelection) {
+            getPicklistDeleteData(model, actions, user)
+                .then(data => {
+                    setNounAndNumber(data.numDeletable === 1 ? '1 Picklist' : data.numDeletable + ' Picklists');
+                    setDeletionData(data);
+                })
+                .catch(() => {
+                    setErrorMessage("There was a problem loading the deletion data. " +
+                        "Verify the " + lcNoun +
+                        (model.selections.size === 1 ? " is still valid and has " : " are still valid and have ")
+                        + "not already been deleted.");
+                });
+        }
+        else {
+            setNounAndNumber('This Picklist');
+            const picklist = new PicklistModel(model.getRow(undefined, true));
+            setDeletionData({
+                numDeletable: 1,
+                numNotDeletable: 0,
+                numShared: picklist.isPublic() ? 1 : 0,
+                deletableLists: [picklist]
+            });
+        }
+    }, [model, actions, user]);
```

Comment:
I wouldn't think so because the setters actually won't ever change during the lifetime of this component, unless I'm not understanding the lifecycle of useState.